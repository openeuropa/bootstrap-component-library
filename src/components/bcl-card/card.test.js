import { renderTwigFileAsNode } from "@openeuropa/bcl-test-utils";
import demoData, { image } from "@openeuropa/bcl-data-card/data";
import demoDataHorizontal from "@openeuropa/bcl-data-card/dataHorizontal";

const template = "@oe-bcl/bcl-card/card.html.twig";
const render = (params, reset) => renderTwigFileAsNode(template, params, reset);

describe("OE - Card", () => {
  test(`renders correctly`, () => {
    expect.assertions(1);

    return expect(render(demoData)).resolves.toMatchSnapshot();
  });

  test(`renders with image bottom correctly`, () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData, image: { ...image, position: "bottom" } }, true)
    ).resolves.toMatchSnapshot();
  });

  test(`renders with variant correctly`, () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData, variant: "danger" }, true)
    ).resolves.toMatchSnapshot();
  });

  test(`renders horizontal correctly`, () => {
    expect.assertions(1);

    return expect(render(demoDataHorizontal)).resolves.toMatchSnapshot();
  });

  test(`renders with border variant correctly`, () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData, border_variant: "danger" }, true)
    ).resolves.toMatchSnapshot();
  });

  test(`renders with text color white correctly`, () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData, text_color: "white" }, true)
    ).resolves.toMatchSnapshot();
  });
});
