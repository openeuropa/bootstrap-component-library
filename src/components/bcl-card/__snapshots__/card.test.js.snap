// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OE - Card renders correctly 1`] = `
<jest>
  <div
    class="w-25 card"
  >
    <div
      class="card-header"
    >
      Header of card
    </div>
    <img
      alt="alt img"
      class="card-img-top"
      src="https://picsum.photos/seed/1002/600/400"
    />
    <div
      class="card-body"
    >
      <h5
        class="card-title"
      >
        Title card
      </h5>
      <h6
        class="card-subtitle mb-2"
      >
        Subtitle card
      </h6>
      <p
        class="card-text mb-2"
      >
        Text card with longer text which is supported
      </p>
      <div
        class="mt-2-5"
      >
        <span
          class="mb-2 me-2 badge rounded-pill badge-outline-primary"
        >
          I'm a badge
        </span>
        <span
          class="test-class mb-2 badge rounded-pill badge-outline-primary"
        >
          I'm also a badge
        </span>
      </div>
    </div>
    <div
      class="card-footer"
    >
      Footer of card
    </div>
  </div>
</jest>
`;

exports[`OE - Card renders correctly with badges 1`] = `
<jest>
  <div
    class="w-25 card"
  >
    <div
      class="card-header"
    >
      Header of card
    </div>
    <img
      alt="alt img"
      class="card-img-top"
      src="https://picsum.photos/seed/1002/600/400"
    />
    <div
      class="card-body"
    >
      <h5
        class="card-title"
      >
        Title card
      </h5>
      <h6
        class="card-subtitle mb-2"
      >
        Subtitle card
      </h6>
      <p
        class="card-text mb-2"
      >
        Text card with longer text which is supported
      </p>
      <div
        class="mt-2-5"
      >
        <span
          class="mb-2 badge bg-primary"
        >
          Card Category
        </span>
      </div>
    </div>
    <div
      class="card-footer"
    >
      Footer of card
    </div>
  </div>
</jest>
`;

exports[`OE - Card renders correctly with custom title attributes 1`] = `
<jest>
  <div
    class="card"
  >
    <div
      class="card-header"
    >
      Header of card
    </div>
    <img
      alt="alt img"
      class="card-img-top"
      src="https://picsum.photos/seed/1002/600/400"
    />
    <div
      class="card-body"
    >
      <h5
        attribute="with a value"
        class="card-title"
        test="value"
      >
        Title card
      </h5>
      <h6
        class="card-subtitle mb-2"
      >
        Subtitle card
      </h6>
      <p
        class="card-text mb-2"
      >
        Text card with longer text which is supported
      </p>
      <div
        class="mt-2-5"
      >
        <span
          class="mb-2 me-2 badge rounded-pill badge-outline-primary"
        >
          I'm a badge
        </span>
        <span
          class="test-class mb-2 badge rounded-pill badge-outline-primary"
        >
          I'm also a badge
        </span>
      </div>
    </div>
    <div
      class="card-footer"
    >
      Footer of card
    </div>
  </div>
</jest>
`;

exports[`OE - Card renders horizontal correctly 1`] = `
<jest>
  <div
    class="w-50 card"
  >
    <div
      class="row g-0"
    >
      <div
        class="col-4"
      >
        <img
          alt="alt img"
          class="card-img-top"
          src="https://picsum.photos/seed/1019/400/600"
        />
      </div>
      <div
        class="col-8"
      >
        <div
          class="card-body"
        >
          <h5
            class="card-title"
          >
            Title card
          </h5>
          <h6
            class="card-subtitle mb-2"
          >
            Subtitle card
          </h6>
          <p
            class="card-text mb-2"
          >
            Text card with longer text which is supported
          </p>
          <button
            class="btn btn-primary"
          >
            Custom action
          </button>
        </div>
      </div>
    </div>
  </div>
</jest>
`;

exports[`OE - Card renders with border variant correctly 1`] = `
<jest>
  <div
    class="card border-danger"
  >
    <div
      class="card-header"
    >
      Header of card
    </div>
    <img
      alt="alt img"
      class="card-img-top"
      src="https://picsum.photos/seed/1002/600/400"
    />
    <div
      class="card-body"
    >
      <h5
        class="card-title"
      >
        Title card
      </h5>
      <h6
        class="card-subtitle mb-2"
      >
        Subtitle card
      </h6>
      <p
        class="card-text mb-2"
      >
        Text card with longer text which is supported
      </p>
      <div
        class="mt-2-5"
      >
        <span
          class="mb-2 me-2 badge rounded-pill badge-outline-primary"
        >
          I'm a badge
        </span>
        <span
          class="test-class mb-2 badge rounded-pill badge-outline-primary"
        >
          I'm also a badge
        </span>
      </div>
    </div>
    <div
      class="card-footer"
    >
      Footer of card
    </div>
  </div>
</jest>
`;

exports[`OE - Card renders with image bottom correctly 1`] = `
<jest>
  <div
    class="card"
  >
    <div
      class="card-header"
    >
      Header of card
    </div>
    <div
      class="card-body"
    >
      <h5
        class="card-title"
      >
        Title card
      </h5>
      <h6
        class="card-subtitle mb-2"
      >
        Subtitle card
      </h6>
      <p
        class="card-text mb-2"
      >
        Text card with longer text which is supported
      </p>
      <div
        class="mt-2-5"
      >
        <span
          class="mb-2 me-2 badge rounded-pill badge-outline-primary"
        >
          I'm a badge
        </span>
        <span
          class="test-class mb-2 badge rounded-pill badge-outline-primary"
        >
          I'm also a badge
        </span>
      </div>
    </div>
    <img
      alt="alt img"
      class="card-img-bottom"
      src="https://picsum.photos/seed/1002/600/400"
    />
    <div
      class="card-footer"
    >
      Footer of card
    </div>
  </div>
</jest>
`;

exports[`OE - Card renders with text color white correctly 1`] = `
<jest>
  <div
    class="card text-white"
  >
    <div
      class="card-header"
    >
      Header of card
    </div>
    <img
      alt="alt img"
      class="card-img-top"
      src="https://picsum.photos/seed/1002/600/400"
    />
    <div
      class="card-body"
    >
      <h5
        class="card-title"
      >
        Title card
      </h5>
      <h6
        class="card-subtitle mb-2"
      >
        Subtitle card
      </h6>
      <p
        class="card-text mb-2"
      >
        Text card with longer text which is supported
      </p>
      <div
        class="mt-2-5"
      >
        <span
          class="mb-2 me-2 badge rounded-pill badge-outline-primary"
        >
          I'm a badge
        </span>
        <span
          class="test-class mb-2 badge rounded-pill badge-outline-primary"
        >
          I'm also a badge
        </span>
      </div>
    </div>
    <div
      class="card-footer"
    >
      Footer of card
    </div>
  </div>
</jest>
`;

exports[`OE - Card renders with variant correctly 1`] = `
<jest>
  <div
    class="card bg-danger text-white"
  >
    <div
      class="card-header"
    >
      Header of card
    </div>
    <img
      alt="alt img"
      class="card-img-top"
      src="https://picsum.photos/seed/1002/600/400"
    />
    <div
      class="card-body"
    >
      <h5
        class="card-title"
      >
        Title card
      </h5>
      <h6
        class="card-subtitle mb-2"
      >
        Subtitle card
      </h6>
      <p
        class="card-text mb-2"
      >
        Text card with longer text which is supported
      </p>
      <div
        class="mt-2-5"
      >
        <span
          class="mb-2 me-2 badge rounded-pill badge-outline-primary"
        >
          I'm a badge
        </span>
        <span
          class="test-class mb-2 badge rounded-pill badge-outline-primary"
        >
          I'm also a badge
        </span>
      </div>
    </div>
    <div
      class="card-footer"
    >
      Footer of card
    </div>
  </div>
</jest>
`;
